### DISCUSSION TITLE
Naming the BlueBuild image UI project

### DISCUSSION THREAD

**Discussion by xynydev** [blue-build/.github#18] [General]
This is about a planned project, which would be a browser-based UI (also available as a desktop app via Tauri), with features like the following being ones that I'd be interested in integrating into the project:
- Listing images, both for base images and end users.
- Creating and modifying custom image repos on GitHub through an UI that augments the `.yml` configuration files.
- Generating ISOs for images and automatically switching (rebasing) to them if installed locally.
- Locally: building images using BlueBuild, publishing to a local registry, or using the built-in functionality to build them into a tar archive and switch (rebase) to that.

I should have a bunch more free time to work on BlueBuild next week, and I'd like to kick off this project with an MVP containing the image creation flow. Having a product name for the project would be great, so I'm setting the ball rolling here to gather some hype and name ideas.   
Some of my ideas below:
- BlueBuild Creator
- BlueBuild Workshop
   - (as it's like Steam Workshop)
 
(I'd fancy having it start with "BlueBuild" as it's our "product offering")    
Here's some AI-generated ideas for the mix too:
- BlueBuild Craft
- BlueBuild Maker
- BlueBuild Studio
- BlueBuild Fusion
- BlueBuild Studio
- BlueBuild Lab
- BlueBuild Composer
- BlueBuild Foundry

I notably left out "Forge", as that's already [used by ublue](https://github.com/ublue-os/forge).

Drop your thoughts down below!

**Comment by fiftydinar** [comment]
Tauri app would bundle bluebuild-cli for local rebasing functionality, right?
Idea sounds great!

BlueBuild Wizard is another name suggestion.

If we decide on making a flatpak & distribute it on FlatHub, we would be called "BlueBuild Organization" or shortened "BlueBuild Org", or something similar. Does that sound good?

I heard that Tauri doesn't have official flatpak support. However, there are some flatpak Tauri apps floating around.
Do you have skills to build Tauri flatpak?

I guess that we would distribute app as flatpak-only, while community would build app for other platforms if they desire, without our support.

**Reply by xynydev** [reply]
> Tauri app would bundle bluebuild-cli for local rebasing functionality, right?

Yeah, as library probably.

> If we decide on making a flatpak & distribute it on FlatHub, we would be called "BlueBuild Organization" or shortened "BlueBuild Org", or something similar. Does that sound good?

Why add the "org" suffix? I don't see that on other orgs on Flathub.

> I heard that Tauri doesn't have official flatpak support. However, there are some flatpak Tauri apps floating around.
Do you have skills to build Tauri flatpak?

This is not a priority, but I'm pretty sure that we could figure something out. I found the links below, which have some discussion. This is also more of a long-term project, and immediately shipping on Flathub wouldn't be a priority, I'd try to dev the browser version first, and make it well-structured and maintainable.

https://github.com/tauri-apps/tauri/issues/3619
https://github.com/tauri-apps/tauri/discussions/4426

**Reply by fiftydinar** [reply]
> Why add the "org" suffix? I don't see that on other orgs on Flathub.

I initially wanted to differentiate BlueBuild name as a a project, so adding Org suffix would make it clearer that BlueBuild is also an organization to the users.

But you're probably right that we don't need that.

> This is also more of a long-term project, and immediately shipping on Flathub wouldn't be a priority, I'd try to dev the browser version first, and make it well-structured and maintainable.

I agree about this.

**Comment by xynydev** [comment]
I have chosen the name Workshop as I like it and no-one else said anything. Lazy concensus etc. etc. There's a repo slowly evolving here. https://github.com/blue-build/workshop

**Reply by fiftydinar** [reply]
Sounds good, can't think of anything better
